code/modules/projectiles/projectile/bullets/sniper.dm

```diff
@@ -1,56 +1,99 @@
-// .50 (Sniper)
+// .50 BMG (Sniper)
 
 /obj/projectile/bullet/p50
-	name =".50 bullet"
+	name =".50 BMG bullet"
 	speed = 0.4
 	range = 400 // Enough to travel from one corner of the Z to the opposite corner and then some.
 	damage = 70
 	paralyze = 100
 	dismemberment = 50
+	catastropic_dismemberment = TRUE
 	armour_penetration = 50
-	var/breakthings = TRUE
+	///Determines object damage.
+	var/object_damage = 80
+	///Determines how much additional damage the round does to mechs.
+	var/mecha_damage = 10
 
 /obj/projectile/bullet/p50/on_hit(atom/target, blocked = 0)
-	if(isobj(target) && (blocked != 100) && breakthings)
-		var/obj/O = target
-		O.take_damage(80, BRUTE, BULLET, FALSE)
+	if(isobj(target) && (blocked != 100) && object_damage)
+		var/obj/thing_to_break = target
+		var/damage_to_deal = object_damage
+		if(ismecha(thing_to_break) && mecha_damage)
+			damage_to_deal += mecha_damage
+		thing_to_break.take_damage(damage_to_deal, BRUTE, BULLET, FALSE)
 	return ..()
 
-/obj/projectile/bullet/p50/soporific
-	name =".50 soporific bullet"
+/obj/projectile/bullet/p50/surplus
+	name =".50 BMG surplus bullet"
 	armour_penetration = 0
-	damage = 0
+	paralyze = 0
 	dismemberment = 0
+	catastropic_dismemberment = FALSE
+
+/obj/projectile/bullet/p50/disruptor
+	name =".50 BMG disruptor bullet"
+	damage_type = STAMINA
 	paralyze = 0
-	breakthings = FALSE
+	dismemberment = 0
+	catastropic_dismemberment = FALSE
+	object_damage = 0
+	mecha_damage = 100
+	var/emp_radius = 2
 
-/obj/projectile/bullet/p50/soporific/on_hit(atom/target, blocked = FALSE)
+/obj/projectile/bullet/p50/disruptor/on_hit(atom/target, blocked = FALSE)
 	if((blocked != 100) && isliving(target))
-		var/mob/living/L = target
-		L.Sleeping(400)
+		var/mob/living/living_guy = target
+		living_guy.Sleeping(40 SECONDS) //Yes, its really 40 seconds of sleep, I hope you had your morning coffee.
+	if(ismecha(target)) //especially good against mechs
+		target.take_damage(mecha_damage, BRUTE, BULLET, FALSE)
```

Comment:
Why did you stick mecha damage in the object damage if youre just gonna override it on the subtypes to manually call it again

Reply:
*shrug*